### DISCUSSION TITLE
raw audio recording of the PineBuds Pro

### DISCUSSION THREAD

**Discussion by tachen-cs** [pine64/OpenPineBuds#69] [Q&A] [answered]
Hi, I am wondering if is it possible to access the raw audio recording of the PineBuds Pro?

Thanks,

**Comment by shymega** [comment, marked as answer]
Moved to Discussion. This is not a bug report.

What do you envision by 'raw'? I imagine you mean something like `.WAV`, but there's always going to be a codec involved of sorts. What is your use-case?

**Reply by tachen-cs** [reply]
Thank you for the quick response! Yes, I am doing some research on audio signal processing and want to access the `.wav` audio wave recorded by the earphone's in-ear microphone, so that I can try my algorithm on the waveform directly.

I am wondering if it is possible to access the audio wave recorded by the feedback microphone of PineBuds Pro? 
BTW, what is the codec processing here? May I ask for the sampling rate of the feedback and feedforward microphone here?

**Reply by Ralim** [reply]
The audio cannot be stored on the bud itself, so you can only access the audio sent over the bluetooth. The codec used for this will depend on what the device paired supports as well. So you would need to change which microphone's data is send back over the bluetooth connection and then record on another device. You will only be able to get one microphone at a time this way afaik.

Sampling rate will depend on what audio sampling rate is being used at the time for sending audio to the buds (if any).

**Reply by tachen-cs** [reply]
Thanks for the reply! May I ask the maximum sampling rate PineBuds Pro supports?

**Reply by shymega** [reply]
I'm not sure myself. It does depend on the context. Best way to find out is to delve into the code...

**Reply by tachen-cs** [reply]
Can you shed some light on how to update the earbud's firmware? Is that possible by plugging in the charging case to my laptop (with earbud inside) to upgrade the earbud?

Thanks!

**Reply by Ralim** [reply]
Please see the main readme: 
https://github.com/pine64/OpenPineBuds#usage
